tests/components/test_battery_sensor.py

```diff
@@ -0,0 +1,76 @@
+"""Tests for the Battery Sensor Classes."""
+from typing import List, Type
+
+from j5.backends import Backend
+from j5.boards import Board
+from j5.components import Component
+from j5.components.battery_sensor import BatterySensor, BatterySensorInterface
+
+
+class MockBatterySensorDriver(BatterySensorInterface):
+    """A testing driver for the led."""
+
+    def get_battery_sensor_voltage(self, board: Board, identifier: int) -> float:
+        """Get the voltage of a battery sensor."""
+        return 5.0
+
+    def get_battery_sensor_current(self, board: Board, identifier: int) -> float:
+        """Get the current of a battery sensor."""
+        return 2.0
+
+
+class MockBatterySensorBoard(Board):
+    """A testing board for the BatterySensor."""
+
+    @property
+    def name(self) -> str:
+        """The name of this board."""
+        return "Testing Battery Sensor Board"
+
+    @property
+    def serial(self) -> str:
+        """The serial number of this board."""
+        return "SERIAL"
+
+    @property
+    def supported_components(self) -> List[Type[Component]]:
+        """List the types of component that this Board supports."""
+        return [BatterySensor]
+
+    def make_safe(self):
```

Comment:
Sorry just spotted one more without a type
```suggestion
    def make_safe(self) -> None:
```